File: pset6/sentimental-readability/readability.py
from cs50 import get_string


# Coleman-Liau index is computed as 0.0588 * L - 0.296 * S - 15.8, where L is the average number of letters per 100 words in the text, 
# and S is the average number of sentences per 100 words in the text.


def print_grade(grade_index):
    if grade_index < 1:
        print('Before Grade 1\n')
    elif grade_index >= 1 and grade_index <= 16:
        print(f'Grade {grade_index}\n')
    else:
        print('Grade 16+\n')


# Coleman-Liau index
def readability(letters, words, sentences):
    L = letters / words * 100
    S = sentences / words * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    rounded_index = round(index)
    return rounded_index


# let CS50 handle this
text = get_string('Text: ')

letters = 0
words = 0
sentences = 0

# checking the type of each character
for word in text:
    for character in word:
        if letters == 1:
            words += 1
        if character.isalpha():
            letters += 1
        elif character.isspace():
            words += 1
        elif character == '.' or character == '?' or character == '!':
            sentences += 1


# find the grade index
grade_index = readability(letters, words, sentences)
print_grade(grade_index)

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


string prompt_text(void);
string parse_each(string str);
void readability(int letters, int words, int sentence);
void print_grade(int index);



int main(void)
{
    string str = prompt_text();
    parse_each(str);
}



string prompt_text(void)
{
    string new_str;
    do
    {
        new_str = get_string("Enter string: ");
    }
    while (strlen(new_str) < 1);
    return new_str;
}


string parse_each(string str) //looks through text for letters, words, sentences & counts
{
    int letters = 0;
    int words = 0;
    int sentence = 0;

    //for(int i = 0, n = strlen(str); n < i; i++) // ask Vibhor about the ; vs ,
    for (int i = 0, n = strlen(str); i < n; i++)
    {
        char chr = str[i];

        if (letters == 1) // count the first letter seen as a word
        {
            words++;
        }

        if ((chr >= 65 && chr <= 90) || (chr >= 97 && chr <= 122)) //check ASCII vals for all alphas
        {
            letters ++;
        }
        else if (chr == 32) // SPACE character
        {
            words ++;
        }
        else if (chr == 33 || chr == 46 || chr == 63) // check for (.) and (!)
        {
            sentence ++;
        }
    }
    readability(letters, words, sentence); // calls upon readability function
    return str;
}

void readability(int letters, int words, int sentence) // calculate 'Readability' based on Coleman-Liau index.
{
    int rounded_index;
    float L = (float)letters / words * 100;
    float S = (float)sentence / words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8; // formula
    rounded_index = roundf(index);
    print_grade(rounded_index);
}


void print_grade(index) //function to print grade results
{
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 1 && index <= 16)
    {
        printf("Grade %i\n", index);
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
}